OpenSim/Analyses/JointReaction.h

```diff
@@ -72,12 +72,12 @@ OpenSim_DECLARE_CONCRETE_OBJECT(JointReaction, Analysis);
         /* Joint reference*/
         const Joint* joint;
         /* The body upon which the force is applied */
-        const PhysicalFrame* appliedOnBody;
+        const Frame* appliedOnBody;
```

Comment:
Based on the inheritance diagram in the doxygen for PhysicalFrame, it looks like the set of objects that are Frames and the set of objects that are PhysicalFrames are equal, so changing the return type of `findBaseFrame()` to PhysicalFrame would be fine but wouldn't be any more restrictive than it is now. It looks like PhysicalOffsetFrame is a type of PhysicalFrame, but a PhysicalOffsetFrame cannot be a "base frame". I can't see a class in the inheritance diagram that includes all potential base frames and excludes all frames that cannot be base frames. Perhaps a fix could be to add a pure virtual `bool PhysicalFrame::canActAsBaseFrame()` on PhysicalFrame requiring each concrete class to declare whether they can be a base frame, and checking the flag here?